sound/soc/msm/msm8x16.c

```diff
@@ -542,16 +540,18 @@ static int lineout_status_put(struct snd_kcontrol *kcontrol,
 	pr_debug("%s: external speaker PA mode:%d\n", __func__, state);
 
 	switch (state) {
+	case 0:
+		schedule_delayed_work(&lineout_amp_disable, msecs_to_jiffies(50));
+		break;
 	case 1:
 		schedule_delayed_work(&lineout_amp_enable, msecs_to_jiffies(50));
 		break;
-	case 0:
-		//schedule_delayed_work(&lineout_amp_disable, msecs_to_jiffies(5));
-		msm8x16_ext_spk_control(0);
-		break;
 	case 2:
 		schedule_delayed_work(&lineout_amp_dualmode, msecs_to_jiffies(50));
 		break;
+	case 3:
```

Comment:
case 0 - headphones are not connected
case 1 - connected 3-pin headphones without a microphone
case 2 - I do not know
case 3 - connected 4-pin headphones with microphone

I looked here /sys/class/switch/h2w/state
No case 3 were noises when booting from connected 4-pin headphones with microphone